project/src/main/java/com/google/sps/servlets/ViewProductsServlet.java

```diff
@@ -89,18 +89,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     if (searchId != null) {
       SearchInfo searchInfo = ServletLibrary.retrieveSearchInfo(datastore, searchId);
-      // TODO: integrate once Phillips finishes product search.
-      // if (searchInfo.getGcsUrl() != null) {
-      //   List <String> productSearchIds = ProductSearchLibrary.productSearch(searchInfo.getGcsUrl(), searchInfo.productCategory);
-      //   List<ProductEntity> imageSearchProducts = new ArrayList<>();
-      //   productSearchIds.forEach(productId->imageSearchProducts.add(ServletLibrary.retrieveProductInfo(datastore, productId));
-      //   Set<ProductEntity> setProducts = new HashSet<>(products);
-      //   List<ProductEntity> newProducts = new ArrayList<>();
-      //   for (ProductEntity product : imageSearchProducts) {
-      //     if (setProducts.contains(product)) newProducts.add(product);
-      //   }
-      //   products = newProducts;
-      // }
+
+      //   TODO: integrate once Phillips finishes product search.
+      if (searchInfo.getGcsUrl() != null) {
+        List <String> productSearchIds = ProductSearchLibrary.getSimilarProductsGcs("cloudberryAllProducts", 
+                                            searchInfo.getProductCategory(), searchInfo.getGcsUrl());
+        List<ProductEntity> imageSearchProducts = new ArrayList<>();
+        productSearchIds.forEach(productId->imageSearchProducts.add(ServletLibrary.retrieveProductInfo(datastore, productId)));
+        Set<ProductEntity> setProducts = new HashSet<>(products);
```

Comment:
This is a little hard to follow with the 3 variables:
- `products`
- `setProducts`
- `newProducts`

It would be helpful to have more descriptive variable names. E.g., `imageSearchProducts` is pretty clear.